### ISSUE TITLE
BUG: to_crs fails with None

### ISSUE THREAD

**Issue by GerbenJdeBoer** [geopandas/geopandas#1881]
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of geopandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example

```python
import geopandas
# copy from https://geopandas.org/docs/reference/api/geopandas.GeoSeries.is_empty.html#geopandas.GeoSeries.is_empty
from shapely.geometry import Point, Polygon
d = {'geometry': [Point(), Point(2, 1), None]}
gdf = geopandas.GeoDataFrame(d, crs="EPSG:4326")
# this fails
gdf.to_crs("EPSG:3857")

```

#### Problem description

None values lead to an error, rather than propagating and returning None.

#### Expected Output

None

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.8.5 (default, Jul 28 2020, 12:59:40)  [GCC 9.3.0]
executable : /usr/bin/python3
machine    : Linux-4.19.128-microsoft-standard-x86_64-with-glibc2.29

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : 3.8.0
GEOS lib   : /usr/lib/x86_64-linux-gnu/libgeos_c.so
GDAL       : 2.4.4
GDAL data dir: /usr/local/lib/python3.8/dist-packages/fiona/gdal_data
PROJ       : 7.2.0
PROJ data dir: /usr/local/lib/python3.8/dist-packages/pyproj/proj_dir/share/proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.8.1
pandas     : 1.1.4
fiona      : 1.8.17
numpy      : 1.19.4
shapely    : 1.7.1
rtree      : None
pyproj     : 3.0.0.post1
matplotlib : 3.3.2
mapclassify: None
geopy      : 2.1.0
psycopg2   : None
geoalchemy2: None
pyarrow    : None

</details>

**Comment by martinfleis** [comment]
Thanks for the report! Can you try updating to geopandas 0.9.0? This was fixed in #1618 and is part of the latest release.

**Comment by GerbenJdeBoer** [comment]
My apologies, indeed fixed.
